sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestFramework/ProjectBuilder.cs

```diff
@@ -49,10 +49,18 @@ private ProjectBuilder(Project project)
         public SolutionBuilder GetSolution() =>
             this.solutionWrapper.Value;
 
-        public Compilation GetCompilation(ParseOptions parseOptions = null) =>
-            parseOptions != null
-                ? Project.WithParseOptions(parseOptions).GetCompilationAsync().Result
-                : Project.GetCompilationAsync().Result;
+        public Compilation GetCompilation(ParseOptions parseOptions = null, CompilationOptions compilationOptions = null)
+        {
+            var project = parseOptions != null
+                ? Project.WithParseOptions(parseOptions)
+                : Project;
+
+            var compilation = project.GetCompilationAsync().Result;
+
+            return compilationOptions == null
+                ? compilation
+                : compilation.WithOptions(compilationOptions);
+        }
```

Comment:
Added the ability to pass in _CompilationOptions_ so the caller can specify e.g. that the assembly is a DLL (to prevent the compile warning about _main method not found_).